:memo: Update DNS configuration instructions (#2447)
platformsh/platformsh-docs: docs/src/domains/steps/subdomains.md

is known as the top-level domain (TLD).
Most Internet applications (such as web browsers) handle TLD specially, such as by restricting certain actions.

For example, a web page at `foo.bar.baz.example.com` can usually set a cookie that's linked to:
- `foo.bar.baz.example.com`,
- `bar.baz.example.com`,
- `baz.example.com`,
- `example.com`
But that web page can *not* set a cookie to all `.com` domains.

That allows a single logical site to be segmented across different subdomains but use a single account login cookie without causing a security risk to all `.com` domains.
Other restrictions apply on TLD's, but cookies are the easiest example.

Aside from true TLD, browser makers have a list of domain suffixes that get the same special handling
called the [Public Suffix List (PSL)](https://publicsuffix.org/).
If you added the `example.com` domain to the PSL,
browsers would refuse to set a cookie on `example.com` from a page at `foo.example.com` but would still accept cookies from a page at `example.com`.

### Subdomain hijacking protection

By default, the subdomain hijacking protection is activated, allowing only one project to use a given domain at a time.

> This security measure is there to prevent a malicious actor from registering a project with `evil.example.com`

Reviewer: :question: Do you really think it was unclear before? What might have been misinterpreted? It seems like extra words to express the same thing.

and no other project can have a domain anywhere in `*.example.com`.
(Multiple subdomains within that same project are perfectly fine.)

> In most cases, that's a desirable added layer of security.

Reviewer: :question: Why take out this sentence? It seemed like a good transition.

> The subdomain hijacking protection can cause issues if you want multiple subdomains from the same organization as separate projects such as for example multiple departments at the same university.

Reviewer: :hammer_and_wrench: The new version seems longer and more complex without adding much clarity. Unless you have specific concerns about the original that I don't see, the most I could see would be getting rid of the parentheses:
```suggestion
But you may run into a problem when you want multiple subdomains from the same organization as separate projects.
For example, multiple departments at the same university.
```

## Locked domains

> In certain cases (such as if your domain was manually added by the support team),

Reviewer: :hammer_and_wrench: I feel like this decreases clarity because it removes the idea of what support team it is. I'm also not sure why the word order was changed, since it seems to me to be the wrong emphasis now.
```suggestion
In certain cases (such as if your domain was added manually by Platform.sh support),
```

## Claimed domains

If you try to use a domain that's claimed, you see an error like the following:

```text
This domain is already claimed by another project. If this is incorrect or you are trying to add a subdomain, please open a ticket with support.
```

> This relates to the [subdomain hijacking prevention](#subdomain-hijacking-protection).

Reviewer: :clipboard: 
```suggestion
This relates to the [subdomain hijacking protection](#subdomain-hijacking-protection).
```
